backend/stubs/structs.pyi

```diff
@@ -13,3 +13,10 @@ class UserInfo:
     def serialize(
         self: 'UserInfo', skip_list: Optional[List[str]], serialize_method: Callable[[Dict[Any, Any]], T]
     ) -> T: ...
+
+    @overload
+    def deserialize(self, serialized_obj: Mapping[bytes, bytes], from_redis: Literal[True]) -> 'UserInfo':
+        ...
+    @overload
+    def deserialize(self, serialized_obj: Mapping[str, Any], from_redis: Literal[False]) -> 'UserInfo':
+        ...
```

Comment:
Confirming my attempt at understanding: you removed the typing of `deserialize` in backend/structs.py because it can take multiple types. the `...` means that there is no special implementation for the typing of this particular overload, so just use the one that's defined. Is this correct?

Reply:
Yea, pretty much :+1: 

Basically the type of one argument depends on the other (I know serialized obj must be bytes if it came from redis; if it didnt, I cant make as strong a statement).
Mypy isn't powerful enough to express this in one line (tbh idk if anything is.  In haskell or TS youd do basically the same as this.  Idk what Rust would look like.)  So what it lets you do is make a stub file, and also this overload decorator exists.

A stub file is an alternative to having types in your application code in general (either because you just dont want to clutter it up OR because you are adding types for code you dont control - some library/dependency that isn't typed).  These are basically meant to be header files, with just the signature, but they have to parse as valid python code.  So you can put anything in the body - convention is to put `...` or `pass`

The overload decorator lets mypy know that there are N valid type signatures for this function